### DISCUSSION TITLE
Timeline for charts

### DISCUSSION THREAD

**Discussion by MiguelLaforest** [livewire/flux#635] [Feature Requests] [answered]
The main feature I was looking forward to, at launch, is the charts (pretty much why I purchased Flux to begin with, other than wanting to support Caleb and Livewire), close second is the date picker.

Long story short, we are building a reports center and someone on the team made the mistake to tell our customers that we'd have pretty, dynamic charts everywhere, for every report. Turns out I find charts really annoying to implement with Livewire, so I've been putting them off for a while and the announcement of Flux with charts came at the perfect time, but if they aren't planned to come out before months (or coming at all?) I might have to bite the bullet and do them myself *\*dies a little bit inside\**.

So that's my thing, but other people probably have other features they are eagerly waiting for and wonder where they are on the production line. While I know it wasn't necessarily sustainable, I liked the daily dev streams on X at the beginning, or simply Caleb being more present there and see a little bit what was going on.

As always, thanks for the hard work, and love Flux with or without charts :)!

**Comment by joshhanley** [comment]
@MiguelLaforest I wasn't aware charts was announced at launch. Have you got a reference for that?

**Reply by MiguelLaforest** [reply]
Maybe "at" launch isn't the right time frame, but I know it was early on. Also "announced" might be a little strong of a term, but Caleb did send an email with a list of upcoming components with charts in it alongside date picker, file upload, and others, if I recall correctly. Unfortunately, I didn't keep the email. During the *Mosty Technical*, right after launch, a few people in chat (including myself) asked about charts, so I guess I'm not the only one that saw that... but the discussion was more focused on the businessy side of Flux, so we didn't get an answer.

All in all Caleb didn't promise charts, but it got mentioned here and there, and I got hooked on that, ahah.

**Reply by joshhanley** [reply]
@MiguelLaforest ah right yep, yeah they talked about charts in the latest Mostly Technical (one with Caleb and Ian) if you want details, but TLDR is charts are hard and Caleb's not sure how to implement them 😆

**Reply by MiguelLaforest** [reply]
Yeeah, saw it a few days after I started this discussion. Makes me happy they both agree charts are difficult and it always sucks to integrate them, ahah. Might start a discussion focused on charts, and see if people have ideas or what they'd be expecting. But I agree, there are a lot of things left to do and charts are a low priority for many.

**Comment by peterjthomson** [comment]
@MiguelLaforest We've worked hard to get a TALL stack implementation for Chart JS that's pretty battle hardened now (multiple charts of different types with different livewire controls on the same page required quite a bit of Alpine futzing). But it's all working now and should play nice with Flux: https://github.com/icehouse-ventures/laravel-chartjs

**Reply by MiguelLaforest** [reply]
Yes, I saw it not long before Flux announcement, actually. I took a look at the demo app and I'm pretty sure it's the package I'll use, I was waiting to see what Caleb might come up with before committing. To be fair, I don't think I was expecting/wanting much more than what yours or some other packages I've seen provide, just a Chart.js wrapper that integrates well with Alpine and Livewire, at least to begin with.

**Comment by EriksonEnterprises** [comment]
I hate charts, including tables, like crazy. Wish someone would come up with the answer to that because I'm not smart enough haha

I usually just go to cards or making custom views of data like a list... but it just looks... Trashy

**Comment by jsmit99** [comment]
Charts would be "the" thing which would make it complete.

Really hope the legend @calebporzio can hook us up!

**Comment by peterjthomson** [comment]
I'm confident that Caleb will kick ass with Charts and the Flux native ones will be awesome. But in the meantime, if you're looking for Livewire compatible charts that accept Laravel style nested arrays (such as from models, etc), then we adopted the largest Chart JS Laravel package and upgraded it to work with Livewire: https://github.com/icehouse-ventures/laravel-chartjs

**Comment by joshhanley** [comment, marked as answer]
Charts has been released in Flux v2.0 so going to close this discussion.
